File: content-solutions/dp/berry-market.java
class Solution {
    public int solution(int[] prices) {
        long hold = Long.MIN_VALUE / 4, sold = 0, rest = 0;
        for (int p : prices) {
            long nextHold = Math.max(hold, rest - p);
            long nextSold = hold + p;
            long nextRest = Math.max(rest, sold);
            hold = nextHold;
            sold = nextSold;
            rest = nextRest;
        }
        return (int) Math.max(sold, rest);
    }
}

File: content-solutions/dp/berry-market.js
function solution(prices) {
  let hold = -Infinity;
  let sold = 0;
  let rest = 0;
  for (const p of prices) {
    const nextHold = Math.max(hold, rest - p);
    const nextSold = hold + p;
    const nextRest = Math.max(rest, sold);
    hold = nextHold;
    sold = nextSold;
    rest = nextRest;
  }
  return Math.max(sold, rest);
}
